web/src/peer_data.ts

```diff
@@ -167,6 +219,14 @@ export function is_user_subscribed(stream_id: number, user_id: number): boolean
     return subscribers.has(user_id);
 }
 
+export async function maybe_fetch_is_user_subscribed(
+    stream_id: number,
+    user_id: number,
+): Promise<boolean> {
+    const subscribers = await get_full_user_set(stream_id);
+    return subscribers.has(user_id);
+}
```

Comment:
I think this is a place where we'll be able to improve considerably through maintaining a set of users for whom we have complete data as well -- if we have most of the active users, very often that set will allow the `participants` loop queries to return instantly without waiting for this fetch.